File: backend/loginTest.js
const http = require('http');

const data = JSON.stringify({
  email: 'admin@santeconnect.sn',
  password: 'thies2024'
});

const options = {
  hostname: '127.0.0.1',
  port: 8000,
  path: '/api/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
    'Content-Length': Buffer.byteLength(data)
  }
};

const req = http.request(options, (res) => {
  console.log('status', res.statusCode);
  let body = '';
  res.on('data', (chunk) => body += chunk);
  res.on('end', () => {
    console.log('body', body);
  });
});

req.on('error', (err) => {
  console.error('error', err.message);
});

req.write(data);
req.end();

File: backend/login_test.py
import json
import urllib.request

url = 'http://127.0.0.1:8000/api/login'
headers = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
}

data = json.dumps({
    'email': 'admin@santeconnect.sn',
    'password': 'thies2024'
}).encode('utf-8')

req = urllib.request.Request(url, data=data, headers=headers, method='POST')
try:
    with urllib.request.urlopen(req) as res:
        print('status', res.status)
        content = res.read().decode('utf-8')
        print(content)
except urllib.error.HTTPError as e:
    print('status', e.code)
    print(e.read().decode('utf-8'))
except Exception as err:
    print('error', err)
